webopenface/api_views.py

```diff
@@ -0,0 +1,311 @@
+parser.add_argument('--imgDim', type=int,
+                    help="Default image dimension.", default=96)
+parser.add_argument('--cuda', action='store_true')
+parser.add_argument('--unknown', type=bool, default=False,
+                    help='Try to predict unknown people')
+
+args = parser.parse_known_args()
+#
+align = openface.AlignDlib(args[0].dlibFacePredictor)
+net = openface.TorchNeuralNet(args[0].networkModel, imgDim=args[0].imgDim,
+                              cuda=args[0].cuda)
+
+# unknownImgs = np.load("./openface/examples/web/unknown.npy")
+training = False
+svm = None
+
+images = {}
+people = ()
+open_face = None
+
+
+def face(rep, identity):
+    return "{{id: {}, rep[0:5]: {}}}".format(
+        str(identity),
+        rep[0:5]
+    )
+
+@csrf_exempt
+def on_message(request):
+    global open_face
+
+    global training
+    if request.method == 'POST':
+        msg = json.loads(request.body)
+        respons_data = {}
+        if msg['type'] == 'ADD_PERSON':
+            respons_data['type'] = 'ADD_PERSON'
+            print(msg['val'])
+            if msg['val'] != '':
+                p = Person(name=msg['val'])
+                p.save()
+                respons_data['type'] = 'ADD_PERSON'
+                respons_data['id'] = p.pk
+        elif msg['type'] == 'TRAINING':
+            training = msg['val']
+        elif msg['type'] == 'FRAME':
+            respons_data = open_face.processFrame(msg['dataURL'], msg['identity'], msg['training'])
+            print 'cos'
+            # respons_data['type'] = "ANNOTATED"
+            # respons_data['content'] = msg['dataURL']
+        elif msg['type'] == 'GET_FRAME':
+            detected_people = []
+            latest_frame = Frame.objects.latest('add_date')
+            published_recently = latest_frame.was_published_recently()
+            for face in latest_frame.detectedface_set.all():
+                for detected_person in face.detectedpeople_set.all():
+                    detected_people.append(detected_person.person.name)
+            respons_data['type'] = "PREVIEW_FRAME"
+            respons_data['dataURL'] = latest_frame.frame
+            respons_data['publishedRecently'] = published_recently
+            respons_data['detectedPeople'] = detected_people
+            print
+        else:
+            print("Warning: Unknown message type: {}".format(msg['type']))
+
+        return HttpResponse(json.dumps(respons_data), content_type="application/json")
+    preview_list = DetectedPeople.objects.order_by('face__frame__add_date')
+    context = {'preview_list': preview_list}
+    return render(request, 'webopenface/base.html', context)
+
+
+class Face:
+
+    def __init__(self, rep, identity):
+        self.rep = rep
+        self.identity = identity
+
+    def __repr__(self):
+        return "{{id: {}, rep[0:5]: {}}}".format(
+            str(self.identity),
+            self.rep[0:5]
+        )
+
+
+class OpenFaceServerProtocol:
```

Comment:
Classes which are not a views should be moved to separate file. This file should be mainly for views.